### PR TITLE
Build wheels with Python 3.10

### PR DESCRIPTION
**Pull request by armenzg**
This will generate the cp310 wheels.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ jobs:
       - uses: actions/checkout@v4
       - run: rustup toolchain install stable --profile minimal --no-self-update
       - uses: actions/setup-python@v5
+        with:
+            python-version: 3.10
```
Comment (asottile-sentry) [inline .github/workflows/build.yml:17]: oh I see, this was using the `.python-version` file before -- maybe just revert the change to that that bumped it to 3.11?

File: .python-version [hunk 2]
```diff
@@ -1 +1 @@
-3.11.8
+3.10.11
```
Comment (armenzg) [inline .python-version:1]: From [here](https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json) we can see that it has an arm64 build: `python-3.10.11-darwin-arm64.tar.gz`

**Review by asottile-sentry — APPROVED** [review]
